README.rst

```diff
@@ -411,6 +536,56 @@ This is the list of all patches used across tutor-mfe (outside of any plugin). A
     cd tutor-mfe
     git grep "{{ patch" -- tutormfe/templates
 
+mfe-env-config-buildtime-imports
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Use this patch for any static imports you need in ``env.config.jsx``. They will be available here if you used the `mfe-docker-post-npm-install patch <#mfe-docker-post-npm-install>`_ to install an NPM package for all MFEs.
+
+It gets rendered at the very top of the file. You should use normal `ES6 import syntax <https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/import>`_.
+
+Note that if you want to only import a module for a particular MFE, doing it here won't work: you'll probably want to use the ``mfe-env-config-runtime-definitions-{}`` patch described below.
+
+File changed: ``tutormfe/templates/mfe/build/mfe/env.config.jsx``
+
+mfe-env-config-buildtime-definitions
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Use this patch for arbitrary ``env.config.jsx`` javascript code that gets evaluated at build time. It is particularly useful for defining slightly more complex components for use in plugin slots.
+
+There's no version of this patch that runs per MFE.  If you want to define MFE-specific code, you should use the MFE-specific ``mfe-env-config-runtime-definitions-{}`` to achieve the same effect.
+
+File changed: ``tutormfe/templates/mfe/build/mfe/env.config.jsx``
+
+mfe-env-config-runtime-definitions
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This patch gets rendered inside an ``async`` function in ``env.config.jsx`` that runs in the browser, allowing you to define conditional imports for external modules that may only be available at runtime. Just make sure to use `import() function <https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/import>`_ syntax:
+
+.. code-block:: javascript
+
+    const mymodule1 = await import('mymodule1');
+    const { mycomponent1, mycomponent2 } = await import('mymodule2');
+
+Note that if any module is not available at runtime, ``env.config.jsx`` execution will fail silently.
```

Comment:
Could we update the documentation to specify that if a component is exported as a default export, it should be accessed using Component.default when dynamically imported? This clarification will help ensure proper usage and avoid potential errors during runtime.

I was testing with indigo and here's my file:
```
// import Footer from '@edly-io/indigo-frontend-component-footer';

...

if (process.env.APP_ID == 'learner-dashboard') {
const Footer = await  import('@edly-io/indigo-frontend-component-footer');
...
{
      op: PLUGIN_OPERATIONS.Insert,
      widget: {
        id: 'default_contents',
        type: DIRECT_PLUGIN,
        priority: 1,
        RenderWidget: <Footer />,
      },
    },
...
}

```

It is throwing error `Element type is invalid: expected a string (for built-in components) or a class/function (for composite components) but got: object.` 

If I use `const { Footer } = await  import('@edly-io/indigo-frontend-component-footer');`, it throws this error `Element type is invalid: expected a string (for built-in components) or a class/function (for composite components) but got: undefined. You likely forgot to export your component from the file it's defined in, or you might have mixed up default and named imports.`.

The footer is exported as default. `export default Footer;`. The errors are generated when I shift to dynamic import `await (...)`. Could we add in doc that we have to use `Footer(Component).default` in case of dynamic imports.

Reply:
Sure, I'll point to the MDN documentation.  They mention that you have to use [the `default` key](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/import#module_namespace_object).

Reply:
They actually document a better way [further down in the page](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/import#importing_defaults), which is what I ended up putting up as an example.